src/executor.rs

```diff
@@ -40,7 +40,8 @@ mod platform {
         }
 
         fn enter<R>(&self, f: impl FnOnce() -> R) -> R {
-            self.0.enter(f)
+            let _guard = self.0.enter();
+            f()
```

Comment:
I think this will not work with other executor implementations. Instead, it should probably be:

```suggestion
            Executor::enter(&self.0, f)
```